code/modules/hydroponics/grown/melon.dm

```diff
@@ -39,6 +40,36 @@
 /obj/item/food/grown/watermelon/make_dryable()
 	return //No drying
 
+/obj/item/food/grown/watermelon/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/kitchen/spoon))
+		return ..()
+
+	var/melon_pulp_count = 1
+	if(seed)
+		melon_pulp_count += round(seed.potency / 25)
+
+	user.balloon_alert(user, "scooped out [melon_pulp_count] pulp")
+	for(var/i in melon_pulp_count)
+		new /obj/item/food/watermelonmush(user.loc)
+
+	var/melon_armour
+	var/melon_chestplate_chance = (max(0, seed.potency - 50) / 50)
```

Comment:
note: don't need to wrap the math here in (), it's an entirely stylistic decision that's up to you. just in case you were not aware